xgcm/grid_ufunc.py

```diff
@@ -39,6 +41,16 @@
 _SIGNATURE = f"^{_ARGUMENT_LIST}->{_ARGUMENT_LIST}$"
 
 
+def _maybe_unpack_vector_component(da):
+    if isinstance(da, dict):
+        da_component = da[
+            list(da.keys())[0]
+        ]  # we only allow len 1 dicts previously. Not sure if this is the most elegant way to do this
+    else:
+        da_component = da
+    return da_component
```

Comment:
```suggestion
def _maybe_unpack_vector_component(data: Union[DataArray, dict[str, DataArray]]) -> DataArray:
    if isinstance(data, dict):
        [da] = list(data)  # this will raise if more than one element
        return da
    else:
        return da
```